aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -448,6 +452,30 @@ Tensor norm(const Tensor& self, Scalar p) {
   return at::native::_norm(self, p);
 }
 
+inline Tensor & _all(Tensor & result, std::unique_ptr<TensorIterator> & iter) {
+  if (iter->numel() == 0) {
+    result.fill_(1);
+  } else {
+    and_stub(iter->device_type(), *iter);
+  }
+
+  return result;
+}
+
+Tensor all(const Tensor& self) {
+  AT_CHECK(self.type().backend() == Backend::CPU ||
+    self.type().backend() == Backend::CUDA, "all only supports CPU AND CUDA "
+    "backend, got: ", toString(self.type().backend()));
+  AT_CHECK(self.type().scalarType() == at::ScalarType::Byte,
+    "all only supports torch.uint8 dtype");
+
+  Tensor result = at::empty({0}, self.options());
+  ScalarType dtype = get_dtype(result, self, {}, true);
```

Comment:
why call get_dtype?  You always know the answer is going to be result.dtype.  The arguments to this also don't look correct; you don't want to promote_integers for all -- there's no risk of overflow.